selfdrive/registration.py

```diff
@@ -6,20 +6,19 @@
 from selfdrive.version import version, terms_version, training_version, get_git_commit, get_git_branch, get_git_remote
 from common.android import get_imei, get_serial, get_subscriber_info
 from common.api import api_get
-from common.params import Params
+from common.params import Params, put_nonblocking
 from common.file_helpers import mkdirs_exists_ok
 from common.basedir import PERSIST
 
 def register():
-  params = Params()
-  params.put("Version", version)
-  params.put("TermsVersion", terms_version)
-  params.put("TrainingVersion", training_version)
+  put_nonblocking("Version", version)
+  put_nonblocking("TermsVersion", terms_version)
+  put_nonblocking("TrainingVersion", training_version)
 
-  params.put("GitCommit", get_git_commit(default=""))
-  params.put("GitBranch", get_git_branch(default=""))
-  params.put("GitRemote", get_git_remote(default=""))
-  params.put("SubscriberInfo", get_subscriber_info())
+  put_nonblocking("GitCommit", get_git_commit(default=""))
```

Comment:
Don't use asynchronous version in this file. It could cause a race condition where loggerd starts without the params set